opensearch-operator/pkg/helpers/helpers.go

```diff
@@ -159,3 +164,70 @@ func DiffSlice(leftSlice, rightSlice []string) []string {
 	}
 	return diff
 }
+
+// Count the number of PVCs created for the given NodePool
+func CountPVCsForNodePool(ctx context.Context, k8sClient client.Client, cr *opsterv1.OpenSearchCluster, nodePool *opsterv1.NodePool) (int, error) {
+	clusterReq, err := labels.NewRequirement(ClusterLabel, selection.Equals, []string{cr.ObjectMeta.Name})
+	if err != nil {
+		return 0, err
+	}
+	componentReq, err := labels.NewRequirement(NodePoolLabel, selection.Equals, []string{nodePool.Component})
+	if err != nil {
+		return 0, err
+	}
+	selector := labels.NewSelector()
+	selector = selector.Add(*clusterReq, *componentReq)
+	list := corev1.PersistentVolumeClaimList{}
+	if err := k8sClient.List(ctx, &list, &client.ListOptions{LabelSelector: selector}); err != nil {
+		return 0, err
+	}
+	return len(list.Items), nil
+}
+
+// Delete a STS with cascade=orphan and wait until it is actually deleted from the kubernetes API
```

Comment:
what is STS with cascade ?

Reply:
`cascade=orphan` means kubernetes will not delete the pods of the statefulset. That way any remaining pods keep running and are picked up again as soon as we recreate the STS.